core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceWorkerGroupPerRouteTest.java

```diff
@@ -0,0 +1,153 @@
+                    return super.next();
+                }
+            };
+
+    private static final EventLoopGroup defaultExecutor =
+            new DefaultEventLoopGroup(1) {
+                @Override
+                public EventLoop next() {
+                    defaultServiceWorkerGroupCount.incrementAndGet();
+                    return super.next();
+                }
+            };
+
+    @BeforeEach
+    void clear() {
+        aServiceWorkerGroupCount.set(0);
+        bServiceWorkerGroupCount.set(0);
+        defaultServiceWorkerGroupCount.set(0);
+    }
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.annotatedService().serviceWorkerGroup(aExecutor, true)
+              .build(new MyAnnotatedServiceA());
+            sb.annotatedService().serviceWorkerGroup(bExecutor, false)
+              .build(new MyAnnotatedServiceB());
+            sb.annotatedService(new MyAnnotatedServiceC());
+
+            sb.serviceWorkerGroup(defaultExecutor, true);
+        }
+    };
+
+    @Test
+    void testOnlyEventLoopWithoutBlockingAnnotation() {
+        final WebClient client = WebClient.of(server.httpUri());
+
+        AggregatedHttpResponse res = client.execute(RequestHeaders.of(HttpMethod.GET, "/a"))
+                                           .aggregate()
+                                           .join();
+        assertThat(res.status()).isSameAs(HttpStatus.OK);
+        assertThat(aServiceWorkerGroupCount).hasValue(1);
+        assertThat(bServiceWorkerGroupCount).hasValue(0);
+        assertThat(defaultServiceWorkerGroupCount).hasValue(0);
+
+        res = client.execute(RequestHeaders.of(HttpMethod.GET, "/b"))
+                    .aggregate()
+                    .join();
+        assertThat(res.status()).isSameAs(HttpStatus.OK);
+        assertThat(aServiceWorkerGroupCount).hasValue(1);
+        assertThat(bServiceWorkerGroupCount).hasValue(1);
+        assertThat(defaultServiceWorkerGroupCount).hasValue(0);
+
+        res = client.execute(RequestHeaders.of(HttpMethod.GET, "/c"))
+                    .aggregate()
+                    .join();
+        assertThat(res.status()).isSameAs(HttpStatus.OK);
+        assertThat(aServiceWorkerGroupCount).hasValue(1);
+        assertThat(bServiceWorkerGroupCount).hasValue(1);
+        assertThat(defaultServiceWorkerGroupCount).hasValue(1);
+
+        assertThat(aExecutor.isShutdown()).isFalse();
+        assertThat(bExecutor.isShutdown()).isFalse();
+        assertThat(defaultExecutor.isShutdown()).isFalse();
+
+        server.stop().join();
+
+        assertThat(aExecutor.isShutdown()).isTrue();
+        assertThat(bExecutor.isShutdown()).isFalse();
+        assertThat(defaultExecutor.isShutdown()).isTrue();
+    }
+
+    static class MyAnnotatedServiceA {
+        @Get("/a")
+        public HttpResponse httpResponse() {
```

Comment:
Could we add an assertion that verifies the service is invoked from the specified `EventLoop`?